src/main/java/duke/TaskManager.java

```diff
@@ -0,0 +1,375 @@
+
+    public TaskManager() {
+        this.tasks = new ArrayList<>();
+    }
+
+    private int getTasksSize() {
+        return tasks.size();
+    }
+
+    public void loadData() {
+        try {
+            loadSavedData();
+        } catch (FileNotFoundException e) {
+            File file = new File(DATA_FILEPATH);
+            try {
+                file.createNewFile();
+            } catch (IOException ex) {
+                System.out.println("     Unable to create new file");
+            }
+
+        }
+    }
+
+    public void loadSavedData() throws FileNotFoundException {
+        File f = new File(DATA_FILEPATH);
+        Scanner s = new Scanner(f);
+
+        while(s.hasNext()) {
+            loadSavedInputToTasks(s.nextLine());
+        }
+    }
+
+    private void loadSavedInputToTasks(String input) {
+        String[] inputs = input.split(SEPARATOR_VERTICAL_LINE);
+        String taskType = inputs[0];
+        String isTaskDone = inputs[1];
+        String taskDescription = inputs[2];
+
+        if (taskType.equals(INITIAL_TODO)) {
+            addTodoTaskWithException(COMMAND_TODO + " " + taskDescription, FOR_LOADING);
+        } else if (taskType.equals(INITIAL_DEADLINE)) {
+            String taskDeadline = inputs[3];
+            addDeadlineTaskWithException(COMMAND_DEADLINE + " " + taskDescription
+                    + SEPARATOR_BY + taskDeadline, FOR_LOADING);
+        } else {
+            String eventTime = inputs[3];
+            addEventTaskWithException(COMMAND_EVENT + " " + taskDescription
+                    + SEPARATOR_AT + eventTime, FOR_LOADING);
+        }
+
+        if (isTaskDone.equals(TASK_IS_DONE)) {
+            tasks.get(tasks.size() - 1).markAsDone();
+        }
+    }
+
+    public void processUserInput() {
+        Scanner sc = new Scanner(System.in);
+        while(true) {
+            String userInput = sc.nextLine();
+            if (userInput.equals(COMMAND_BYE)) {
+                break;
+            }
+
+            if (userInput.equals(COMMAND_LIST)) {
+                printAllTasks();
+            } else if (userInput.startsWith(COMMAND_DONE)) {
+                String[] inputs = userInput.split(SEPARATOR_SPACE);
+                int taskDoneNumber = Integer.parseInt(inputs[1]);
+
+                setTaskDoneWithException(taskDoneNumber - 1);
+            } else if (userInput.startsWith(COMMAND_DELETE)) {
+                String[] inputs = userInput.split(SEPARATOR_SPACE);
+                int taskDeleteNumber = Integer.parseInt(inputs[1]);
+
+                deleteTaskWithException(taskDeleteNumber - 1);
+            }
+            else {
+                try {
+                    addTask(userInput);
+                } catch (WrongCommandException e) {
+                    printHorizontalLine();
+                    System.out.println("     ☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+                    printHorizontalLine();
+                }
```

Comment:
Abstracting out the code for each type of command would help here. See [SLAP](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard). You have partly done it for `COMMAND_LIST` with `printAllTasks()`.